### ISSUE TITLE
The `doc_server` example doesn't appear to work and has a few issues. 

### ISSUE THREAD

**Issue by nelsonjchen** [iron/staticfile#80]
https://github.com/iron/staticfile/blob/2ac0d6048415bb9e9f4ae169248225df8cb1996b/examples/doc_server.rs#L7

For starters, I'm not sure why we should run `cargo test` to run an example. 

Second, the `doc_server` binary would not be generated from these instructions. Perhaps we meant to say `cargo run --example doc_server`? Was `--example` introduced after this example?

That said, the "doesn't appear to work" comes from the last instruction, "point your browser to http://127.0.0.1:3000/doc/". If I do that, I get a blank page. I get this from cURL request to that endpoint. 

```
$ curl -vvv http://localhost:3000/doc/                                                                           
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 3000 (#0)
> GET /doc/ HTTP/1.1
> Host: localhost:3000
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 404 Not Found
< Date: Thu, 28 Jul 2016 08:12:41 GMT
< Content-Length: 0
<
* Connection #0 to host localhost left intact
```

I do get the `COPYRIGHT.txt` and any other content below the `/` route like http://127.0.0.1:3000/COPYRIGHT.txt, but the `doc`and `src` appear to end with 404s. Am I doing something wrong?

**Comment by Hoverbear** [comment]
According to `git blame` line 7 hasn't been added since sept 2014, so definitely. :)

I'm looking into this.

**Comment by Hoverbear** [comment]
Hey!
I'm afraid I'm not having that problem. Here's what I did:

```rust
cargo doc && cargo run --example doc_server
```

Then I browsed to `localhost:3000` and got this:

![example](https://cloud.githubusercontent.com/assets/130903/17237758/0b3766ba-550a-11e6-8916-4df43ad9efaf.jpg)

Can you please confirm you're experiencing something different? Can you tell me your `rustc` version, and your operating system?

**Comment by Hoverbear** [comment]
7346d72b6120a2f71e6dc6a46c53e79baf1ef7b0 fixed the example commands.

**Comment by nelsonjchen** [comment]
rustc 1.10.0 (cfcb716cf 2016-07-03)

OS X 10.11.6. 

Case Insensitive HFS+

**Comment by Hoverbear** [comment]
Ahh I'm on `rustc 1.12.0-nightly (936bfea94 2016-07-20)`. I'll try an older version...

**Comment by nelsonjchen** [comment]
I can try a newer version too.

**Comment by Hoverbear** [comment]
`rustc 1.10.0 (cfcb716cf 2016-07-03)` seems to work as well. I'm on Linux, do you know anyone who can test on OS X?

**Comment by nelsonjchen** [comment]
I am indeed testing on master.

**Comment by Hoverbear** [comment]
@nelsonjchen Can you confirm the `router` example is also not working for you?

**Comment by nelsonjchen** [comment]
It works. The `hello` route works and ,😆, the off by one letter static file mount route "`doc`**s**" also appears to work.

**Comment by nelsonjchen** [comment]
Hold on, that's with this rust version.

rustc 1.12.0-nightly (feeca9457 2016-07-26)

**Comment by nelsonjchen** [comment]
😕 . Odd, now things seem to work, at least on this MacBook. 

I could have sworn I was doing `cargo doc --open` and seeing docs and then wondering what's happening with `staticfile`'s `doc_server`. I still have another Macbook with the same symptom, but it's a few miles away and a few hours away in schedule for me to get to.

**Comment by nelsonjchen** [comment]
That is with rustc 1.10.0 (cfcb716cf 2016-07-03). Just works. 😖 . I'll close this for now. I'm like 80% sure I'm doing something stupid here. If I can still reproduce it on that other MacBook, I'll be sure to comment.

**Comment by nelsonjchen** [comment]
I left the lid open and I was able to remote into the other Macbook. 

I was able to reproduce this issue. I checked I was running the current stable. I ran `cargo doc` and I got no output, so I guess it was up-to-date. I ran `cargo run --example doc_server`. I checked that I was accessing `http://127.0.0.1/doc/`, without the "s". I still got a blank page. I ran `cargo doc --open` and got the docs. I moved this local repository away.

I made another clone of the repository and repeated the steps but this time when I visited "http://127.0.0.1/doc/", it worked! 😖 

I guess something bad compilation artifacts were affecting things. Maybe I just ran the example at a bad time and kept on building it with the "bad time"'s Lockfile where some dependency broke for a few moments. That would explain why things are working now.

Hmm... I just confirmed it. It was just that. When I use this Cargo.lock, `doc_server` results in a blank page. Here's the lockfile.

https://gist.github.com/nelsonjchen/a7707b9cfac1acb4c8b61748e4087e42

@Hoverbear , thank you for your help!
